packages/sanity/src/_internal/cli/actions/schema/utils/mainfestExtractor.ts

```diff
@@ -0,0 +1,50 @@
+import {type CliCommandArguments, type CliCommandContext, type CliOutputter} from '@sanity/cli'
+import chalk from 'chalk'
+
+import {type ExtractManifestFlags, extractManifestSafe} from '../../manifest/extractManifestAction'
+import {FlagValidationError} from './schemaStoreValidation'
+
+export type ManifestExtractor = (manifestDir: string) => Promise<void>
+
+export async function isManifestExtractSatisfied(args: {
```

Comment:
the naming here feels slightly off. It _sounds_ like a predicate/pure function, but it's doing a lot of heavy lifting. Wondering if the name should better reflect that?

Reply:
Will ensureManifestExtractSatisfied fly?